app/src/main/java/org/httpsknighthacks/knighthacksandroid/Profile.java

```diff
@@ -37,22 +43,32 @@
     private static final String MY_PREFERNCES = "MY_PREF";
     private static final String IMAGE_PROFILE = "image";
 
+    CardView mNotificationSettings;
+    CardView mGroupsActivity;
+
+    Dialog mDialog;
+
     @Override
     protected void onCreate(Bundle savedInstanceState){
         super.onCreate(savedInstanceState);
 
         setContentView(R.layout.activity_profile);
 
-        CardView scanQrCodeButton = findViewById(R.id.profile_scan_qr_code_button);
+        // The comments are need because they contain code to run features
```

Comment:
Actually, none of this code is going to end up being in this activity class. The profile activity only exists to hold the fragments, and each fragment has its own class that handles all the views belonging to that fragment. I've already made these class I just haven't pushed them. I implemented the scanQrCodeButton and as well as the image picker so you don't need to worry about that. You can remove all these functions from the activity but keep them saved somewhere because I will need them when I implement them in the fragment classes.

Reply:
Oh ok